### PR TITLE
#1274 Update icons in common properties

### PR DESCRIPTION
**Pull request by nabilIbm**
Fixes : #1274 
Signed-off-by: nabik <nabil@nabil.com>

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by matthoward366** [comment]
Screen Shot 2022-12-07 at 6 16 47 AM

I don't think you need to set the color as well since this is already being inherited from the button.

File: canvas_modules/common-canvas/src/common-properties/controls/toggletext/toggletext.scss [hunk 1]
```diff
@@ -29,11 +29,19 @@
...
+		height: $spacing-04;
+		padding-left: $spacing-02;
+	}
 }
 
 // toggle control in table needs to fit within column
 .properties-table-cell-control {
 	.properties-toggletext button {
 		padding: 0 $spacing-02;
+		svg {
```
Comment (caritaou) [inline canvas_modules/common-canvas/src/common-properties/controls/toggletext/toggletext.scss:42]: This is not needed as the css for the `.properties-toggletext` also applies to those inside tables

File: canvas_modules/harness/assets/images/down-triangle.svg [hunk 2]
```diff
@@ -1 +1 @@
-<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1024 1024" height="10" width="10"><path d="M42 128l469 810 469-810z"/></svg>
+<svg width="10" height="5" fill-rule="evenodd"><path d="M0 0l5 4.998L10 0z"></path></svg>
```
Comment (caritaou) [inline canvas_modules/harness/assets/images/down-triangle.svg:1]: Why is the height and width set to 10 and 5?

When I downloaded the caret image from [Carbon library](https://carbondesignsystem.com/guidelines/icons/library/), this is the content:

```
<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 24.0.3, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="icon" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 width="32px" height="32px" viewBox="0 0 32 32" style="enable-background:new 0 0 32 32;" xml:space="preserve">
<style type="text/css">
	.st0{fill:none;}
</style>
<title>caret--down</title>
<polygon points="24,12 16,22 8,12 "/>
<rect id="_Transparent_Rectangle_" class="st0" width="32" height="32"/>
</svg>

```

how did you end up with that file?

Comment (nabilIbm) [reply canvas_modules/harness/assets/images/down-triangle.svg:1]: Hi @caritaou , I did grab them from the test harness,  are these files relevant?
image
image

Comment (nmgokhale) [reply canvas_modules/harness/assets/images/down-triangle.svg:1]: @nabilIbm You have to download carbon icons SVGs from here - https://v10.carbondesignsystem.com/guidelines/icons/library
Screen Shot 2022-12-08 at 10 02 47 AM

Comment (caritaou) [reply canvas_modules/harness/assets/images/down-triangle.svg]: @tomlyn Unfortunately that approach will not work for the Toggletext control. We allow applications to pass in custom images to display https://github.com/elyra-ai/canvas/blob/main/canvas_modules/common-canvas/src/common-properties/controls/toggletext/toggletext.jsx#L53.

This issue is to update the svg files used in the examples in the test harness to come from Carbon and to set generic styles in the Toggletext control to align the icons

Comment (tomlyn) [reply canvas_modules/harness/assets/images/down-triangle.svg]: Oh OK, I see, so you don't need Carbon icons for this then you can just use any SVG file we have.

File: canvas_modules/common-canvas/src/common-properties/controls/toggletext/toggletext.scss [hunk 3]
```diff
@@ -29,6 +29,10 @@
 			border-bottom: 2px solid $support-03;
 		}
 	}
+	button svg {
+		padding-left: $spacing-02;
+		padding-top: $spacing-02;
+	}
```
Comment (nmgokhale) [inline canvas_modules/common-canvas/src/common-properties/controls/toggletext/toggletext.scss:35]: You can remove entire `button svg {}` code.
Add following CSS under `.properties-table-cell-control` -

```
// toggle control in table needs to fit within column
.properties-table-cell-control {
	.properties-toggletext button {
		padding: 0 $spacing-02;
	}
	.properties-toggletext svg { // this is new CSS
		margin-left: $spacing-02;
	}
}
```

**Comment by nmgokhale** [comment]
@nabilIbm `up-triangle.svg` and `down-triangle.svg` is used in these 3 files in common-canvas -
Screen Shot 2022-12-23 at 2 42 30 PM

In all these files, update the `x_pos, y_pos, width, height` for all objects using `up-triangle.svg` and `down-triangle.svg`. 
**Existing code on main branch-**
```
{ id: "dec-1", position: "source", image: "images/up-triangle.svg", class_name: "det-tri",
				distance: 40, x_pos: -7, y_pos: -7, width: 14, height: 14, outline: true, tooltip: "Up Triangle", temporary: true },
{ id: "dec-2", position: "target", image: "images/down-triangle.svg", class_name: "det-tri",
				distance: -40, x_pos: -7, y_pos: -7, width: 14, height: 14, outline: true, tooltip: "Down Triangle", temporary: true },
```

**Updated code -**
```
{ id: "dec-1", position: "source", image: "images/up-triangle.svg", class_name: "det-tri",
				distance: 40, x_pos: -16, y_pos: -16, width: 36, height: 36, outline: true, tooltip: "Up Triangle", temporary: true },
{ id: "dec-2", position: "target", image: "images/down-triangle.svg", class_name: "det-tri",
				distance: -40, x_pos: -16, y_pos: -16, width: 36, height: 36, outline: true, tooltip: "Down Triangle", temporary: true },
```

**Comment by nmgokhale** [comment]
Read-only canvas shows square border for icons. Please remove the border -
Screen Shot 2023-01-03 at 10 17 54 AM

**Comment by nabilIbm** [comment]
@nmgokhale can you please take a look
image

**Review by nmgokhale — APPROVED** [review]
Looks good.

**Pull request merged**